eldritch/expansions/seaside/abilities.py

```diff
@@ -0,0 +1,114 @@
+import typing
+
+from eldritch import assets
+from eldritch import characters
+from eldritch import events
+from eldritch import values
+
+if typing.TYPE_CHECKING:
+  from eldritch.eldritch import GameState
+
+
+class AbnormalFocus(assets.Asset):
+  def __init__(self):
+    super().__init__("Abnormal Focus")
+
+  def get_interrupt(self, event, owner: characters.BaseCharacter, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.MoveSliders(owner, {slider + "_slider": 0 for slider in owner.sliders()})
+    return None
+
+
+class BreakingTheLimits(assets.Asset):
+  def __init__(self):
+    super().__init__("Breaking the Limits")
+    self.tokens["sanity"] = 0
+    self.tokens["stamina"] = 0
+
+  def get_interrupt(self, event, owner, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.Sequence([
+          events.RemoveToken(self, "sanity", owner, self.tokens["sanity"]),
+          events.RemoveToken(self, "stamina", owner, self.tokens["stamina"]),
+      ], owner)
+    return None
+
+  def get_usable_interrupt(self, event, owner, state):
+    if (
+        not isinstance(event, events.SliderInput)
+        or event.character != owner
+        or self.get_bonus("abnormal_focus", None, owner, state) >= 3
+        or (owner.stamina == 1 and owner.sanity == 1)
+    ):
+      return None
+    spend = events.SpendChoice(
```

Comment:
I was thinking of attaching the spend to the slider input for slightly fewer clicks in the UI. How are you thinking about making sure the player doesn't overspend?

Reply:
Not sure how to attach the spend to the SliderInput... I wasn't thinking about over-spending (i.e. by spending 2 at first, and then 3 a second time), but an easy work-around is to have the max_spend be `3-sum(self.tokens.values())`

Reply:
Upon further consideration, attaching the spend to the SliderInput seems exceptionally difficult. I think this works, even if it makes it difficult for the player to change their mind about their sliders. If it becomes a problem, we can come back and change it later.